src/tanh.rs

```diff
@@ -0,0 +1,191 @@
+/// Calculate the numerator of the `tanh` approximation.
+fn a(x: f32) -> f32 {
+    let x2 = x * x;
+    (((x2 + 378.) * x2 + 17325.) * x2 + 135135.) * x
+}
+
+/// Calculate the denominator of the `tanh` approximation.
+fn b(x: f32) -> f32 {
+    let x2 = x * x;
+    ((28. * x2 + 3150.) * x2 + 62370.) * x2 + 135135.
+}
+
+/// Compute a fast approximation of the hyperbolic tangent of `x`.
+///
+/// For large |x|, the output may be outside of [-1, 1].
+#[inline]
+pub fn tanh_raw(x: f32) -> f32 {
+    // Implementation based on
+    // https://varietyofsound.wordpress.com/2011/02/14/efficient-tanh-computation-using-lamberts-continued-fraction
+    a(x) / b(x)
+}
+
+/// Compute a fast approximation of the hyperbolic tangent of `x`.
+///
+/// See `atanh_raw` for a faster version that may return incorrect results for
+/// large `|x|` and `nan`.
+#[inline]
+pub fn tanh(x: f32) -> f32 {
```

Comment:
Don't you think clipping might be problematic, because it results in discontinuities?

Reply:
That's a potential problem. An alternative would be to find when the approximation is exactly +/-1, and clip there instead of 4.97 (I think it should be symmetric?), so that the tanh approximation is continuous (although it's derivative won't be).